scripts/bash/s3a_wrapper.sh

```diff
@@ -0,0 +1,174 @@
+#!/bin/bash
+
+# Copyright 2018 ABSA Group Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# We have 2 states.
+#   Write Standardisation
+#   Write Conformance
+# The script will have to figure out which is being run and if it is being written to S3 or HDFS.
+#   In case of standardisation parsing the java properties supplied to figure out the standardisation path.
+#   In case of conformance calling Menas for info over REST API using supplied metadata
+# Then cleaning up the paths to conform to the path schema in query provided.
+
+# TODO - Auto load jceks files
+
+set -e
+
+# Source env variables
+source "$(dirname "$0")/enceladus_env.sh"
+
+# Initialize variables
+keytab=""
+dataset_name=""
+dataset_version=""
+report_date=""
+report_version=""
+
+# The first argument is the name of the original script
+original_script="$(dirname "$0")/$(basename "$1")"
+
+# Shift the first argument so we can process the rest
+shift
+
+# Check if the original script exists
+if [[ ! -f "$original_script" ]]; then
+  echo "Error: The script '$original_script' does not exist in the current directory."
+  exit 1
+fi
+
+# Initialize an array to hold the other arguments
+other_args=()
+
+# Loop through arguments
+while [[ $# -gt 0 ]]; do
+  case "$1" in
+    --menas-auth-keytab)
+      keytab="$2"
+      shift # past argument
+      shift # past value
+      ;;
+    --dataset-name)
+      dataset_name="$2"
+      shift # past argument
+      shift # past value
+      ;;
+    --dataset-version)
+      dataset_version="$2"
+      shift # past argument
+      shift # past value
+      ;;
+    --report-date)
+      report_date="$2"
+      shift # past argument
+      shift # past value
+      ;;
+    --report-version)
+      report_version="$2"
+      shift # past argument
+      shift # past value
+      ;;
+    *)    # unknown option
+      other_args+=("$1") # save it in an array for later
+      shift # past argument
+      ;;
+  esac
+done
+
+# Print the extracted variables
+echo "Keytab: $keytab"
+echo "Dataset Name: $dataset_name"
+echo "Dataset Version: $dataset_version"
+echo "Report Date: $report_date"
+echo "Report Version: $report_version"
+
+# Run klist to check for a current Kerberos ticket
+if klist -s; then
+    echo "Kerberos ticket found."
+else
+    echo "No Kerberos ticket found or ticket is expired. Please run kinit."
+fi
+
+# Get Dataset info
+response=$(curl --negotiate -s -u : "$MENAS_API/dataset/$dataset_name/$dataset_version")
```

Comment:
If no MENAP_API available/running it simply ends without message. No error info. Debug output:`+response=` -- this is last line of script output (`bash -x`).

Reply:
So this is just a wrapper script for a transition period, not an application. It is not something I want to support, promote or figure out every edge case for.